superset-frontend/src/components/Select/Select.test.tsx

```diff
@@ -17,66 +17,467 @@
+
+const type = (text: string) => userEvent.type(getSelect(), text, { delay: 10 });
+
+const open = () => waitFor(() => userEvent.click(getSelect()));
+
+test('displays a header', async () => {
+  const headerText = 'Header';
+  render(<Select {...defaultProps} header={headerText} />);
+  expect(screen.getByText(headerText)).toBeInTheDocument();
+});
+
+test('inverts the selection', async () => {
+  render(<Select {...defaultProps} invertSelection />);
+  await open();
+  userEvent.click(await findSelectOption(OPTIONS[0].label));
+  expect(await screen.findByLabelText('stop')).toBeInTheDocument();
+});
+
+test('displays the selected values first', async () => {
+  render(<Select {...defaultProps} mode="multiple" />);
+  const option3 = OPTIONS[2].label;
+  const option8 = OPTIONS[7].label;
+  await open();
+  userEvent.click(await findSelectOption(option3));
+  userEvent.click(await findSelectOption(option8));
+  await type('{esc}');
+  await open();
+  const sortedOptions = await findAllSelectOptions();
+  expect(sortedOptions[0]).toHaveTextContent(option3);
+  expect(sortedOptions[1]).toHaveTextContent(option8);
+});
+
+test('searches for label or value', async () => {
+  const option = OPTIONS[11];
+  render(<Select {...defaultProps} />);
+  const search = option.value;
+  await type(search.toString());
+  const options = await findAllSelectOptions();
+  expect(options.length).toBe(1);
+  expect(options[0]).toHaveTextContent(option.label);
+});
+
+test('clear all the values', async () => {
+  const onClear = jest.fn();
+  render(
+    <Select
+      {...defaultProps}
+      mode="multiple"
+      value={[OPTIONS[0], OPTIONS[1]]}
+      onClear={onClear}
+    />,
+  );
+  userEvent.click(screen.getByLabelText('close-circle'));
+  expect(onClear).toHaveBeenCalled();
+  const values = await findAllSelectValues();
+  expect(values.length).toBe(0);
+});
+
+test('does not add a new option if allowNewValue is false', async () => {
+  render(<Select {...defaultProps} options={loadOptions} />);
+  await open();
+  await type(NEW_OPTION);
+  expect(await screen.findByText(NO_DATA)).toBeInTheDocument();
+});
+
+test('static - renders the select with default props', () => {
+  render(<Select {...defaultProps} />);
+  expect(getSelect()).toBeInTheDocument();
+});
+
+test('static - opens the select without any data', async () => {
+  render(<Select {...defaultProps} options={[]} />);
+  await open();
+  expect(screen.getByText(NO_DATA)).toBeInTheDocument();
+});
+
+test('static - makes a selection in single mode', async () => {
+  render(<Select {...defaultProps} />);
+  const optionText = 'Emma';
+  await open();
+  userEvent.click(await findSelectOption(optionText));
+  expect(await findSelectValue()).toHaveTextContent(optionText);
+});
+
+test('static - multiple selections in multiple mode', async () => {
+  render(<Select {...defaultProps} mode="multiple" />);
+  await open();
+  const firstOption = OPTIONS[0];
+  const secondOption = OPTIONS[1];
```

Comment:
nit: some of these might look nicer by doing the ol'
```suggestion
  const [firstOption, secondOption] = OPTIONS;
```